tests/Unit/ParallelAlgorithms/LinearSolver/Gmres/Test_DistributedGmresPreconditionedAlgorithm.yaml

```diff
@@ -0,0 +1,55 @@
+# Distributed under the MIT License.
+# See LICENSE.txt for details.
+
+# The test problem being solved here is a DG-discretized 1D Poisson equation
+# -u''(x) = f(x) on the interval [0, pi] with source f(x)=sin(x) and homogeneous
+# Dirichlet boundary conditions such that the solution is u(x)=sin(x) as well.
```

Comment:
For my education (so, just looking for a Github reply), would I expect to be able to solve this problem without the preconditioning? It appears to be a relatively simple setup, to someone who doesn't know much about elliptic solves.

Reply:
Oh yes, it's very simple. It has only 6 DOFs, so even the non-preconditioned GMRES will solve it in at most 6 iterations, and actually needs only 3. You can see in this input file that by running 65 Richardson-iterations in every GMRES iteration the number of GMRES iterations reduces from 3 to 2. This reduction in iterations is of course not worth the additional cost of running the Richardson iterations at all. But to invert a 6x6 matrix we don't even need an iterative solver, any direct solver could do that. Real elliptic problems that we have involve millions of DOFs and those can't be solved by GMRES alone. They need preconditioning that (1) parallelizes and (2) keeps the number of GMRES iterations approximately constant when increasing the problem size, i.e. the parallelizable work of the preconditioner should scale with size, not the number of GMRES iterations. So instead of the Richardson solver we'll have a Multigrid solver combined with a Schwarz subdomain smoother. Both will slowly make their way into spectre in upcoming PRs.

Reply:
That's helpful, thank you!